crates/hulk_behavior_simulator/src/bin/ingame_penalty_kick_opponent.rs

```diff
@@ -0,0 +1,56 @@
+use bevy::prelude::*;
+
+use scenario::scenario;
+use spl_network_messages::{GameState, PlayerNumber, SubState, Team};
+
+use hulk_behavior_simulator::{
+    game_controller::{GameController, GameControllerCommand},
+    robot::Robot,
+    time::{Ticks, TicksTime},
+};
+
+#[scenario]
+fn ingame_penalty_kick_opponent(app: &mut App) {
+    app.add_systems(Startup, startup);
+    app.add_systems(Update, update);
+}
+
+fn startup(
+    mut commands: Commands,
+    mut game_controller_commands: EventWriter<GameControllerCommand>,
+) {
+    for number in [
+        PlayerNumber::One,
+        PlayerNumber::Two,
+        PlayerNumber::Three,
+        PlayerNumber::Four,
+        PlayerNumber::Five,
+        PlayerNumber::Six,
+        PlayerNumber::Seven,
+    ] {
+        commands.spawn(Robot::new(number));
+    }
+    game_controller_commands.send(GameControllerCommand::SetGameState(GameState::Ready));
+}
+
+fn update(
+    game_controller: ResMut<GameController>,
+    mut game_controller_commands: EventWriter<GameControllerCommand>,
+    time: Res<Time<Ticks>>,
+    mut exit: EventWriter<AppExit>,
+) {
+    if time.ticks() == 3000 {
+        game_controller_commands.send(GameControllerCommand::SetSubState(
+            Some(SubState::PenaltyKick),
+            Team::Opponent,
+        ));
+    }
+    if game_controller.state.hulks_team.score > 0 {
+        println!("Done");
+        exit.send(AppExit::Success);
+    }
+    if time.ticks() >= 10_000 {
+        println!("No goal was scored :)");
+        exit.send(AppExit::Success);
+    }
```

Comment:
Why are both of these success cases? Is that intended?

Reply:
No this was poorly thought through. In some scenarios the goal doesn't make since as a finish condition because they are purely meant for evaluation (such as walk around ball). In the case of the penalty kicks I have no updated this to check for the opponent scoring.